homeassistant/components/lektrico/sensor.py

```diff
@@ -0,0 +1,294 @@
+        """Initialize a Lektrico Device."""
+        self._device = device
+        self._hass = hass
+        self.friendly_name = friendly_name.replace(" ", "_")
+        self._name = friendly_name
+        self._coordinator: DataUpdateCoordinator
+        self._update_fail_count = 0
+        self._info = None
+
+    @property
+    def coordinator(self) -> DataUpdateCoordinator:
+        """Return the coordinator of the Lektrico device."""
+        return self._coordinator
+
+    async def init_device(self) -> bool:
+        """Init the device status and start coordinator."""
+
+        # Create status update coordinator
+        await self._create_coordinator()
+
+        return True
+
+    async def async_device_update(self) -> lektricowifi.Info:
+        """Async Update device state."""
+        a_data = self._hass.async_add_job(self._device.charger_info)
+        data = await a_data
+        entity_reg = er.async_get(self._hass)
+        my_entry = entity_reg.async_get(f"sensor.{self.friendly_name}_{SENSORS[0].key}")
+        if my_entry is not None:
+            dev_reg = dr.async_get(self._hass)
+            if my_entry.device_id is not None:
+                device = dev_reg.async_get(my_entry.device_id)
+                if device is not None:
+                    dev_reg.async_update_device(device.id, sw_version=data.fw_version)
+        return data
+
+    async def _create_coordinator(self) -> None:
+        """Get the coordinator for a specific device."""
+        coordinator = DataUpdateCoordinator(
+            self._hass,
+            _LOGGER,
+            name=f"{DOMAIN}-{self._name}",
+            update_method=self.async_device_update,
+            # Polling interval. Will only be polled if there are subscribers.
+            update_interval=SCAN_INTERVAL,
+        )
+        await coordinator.async_refresh()
+
+        if not coordinator.last_update_success:
+            raise ConfigEntryNotReady
+
+        self._coordinator = coordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Lektrico charger based on a config entry."""
+    charger: lektricowifi.Charger = hass.data[DOMAIN][entry.entry_id]
+
+    await charger.charger_info()
+    settings = await charger.charger_config()
+    _lektrico_device = LektricoDevice(charger, hass, entry.data[CONF_FRIENDLY_NAME])
+    if not await _lektrico_device.init_device():
+        _LOGGER.error("Error initializing Lektrico Device. Name: 1P7K")
+
+    sensors = []
+
+    sensors.extend(
+        [
+            LektricoSensor(
+                charger,
+                sensor_desc,
+                settings,
+                _lektrico_device,
+                entry.data[CONF_FRIENDLY_NAME],
+            )
+            for sensor_desc in SENSORS
+        ]
+    )
+
+    async_add_entities(sensors, True)
```

Comment:
We should not need to update on add when using a data update coordinator

Reply:
I changed to False. Thank you.